### PR TITLE
migrate formatDistanceStrict to TS

### PR DESCRIPTION
**Pull request by a-korzun**
Fixes https://github.com/date-fns/date-fns-typescriptify/issues/33

### CODE REVIEW

File: src/formatDistanceStrict/index.ts [hunk 1]
```diff
@@ -155,37 +156,41 @@ var MINUTES_IN_YEAR = MINUTES_IN_DAY * 365
...
 export default function formatDistanceStrict(
-  dirtyDate,
-  dirtyBaseDate,
-  dirtyOptions
-) {
+  dirtyDate: Date | number,
+  dirtyBaseDate: Date | number,
+  options: LocaleOptions & {
+    addSuffix?: boolean,
+    unit?: 'second' | 'minute' | 'hour' | 'day' | 'month' | 'year',
```
Comment (tan75) [inline src/formatDistanceStrict/index.ts:170]: Could you please move this to [types.ts](https://github.com/date-fns/date-fns/blob/master/src/types.ts) and import it from that file?
see example below:
```
export type Unit =
  | 'year'
  | 'quarter'
  | 'month'
  | 'week'
  | 'day'
  | 'hour'
  | 'minute'
  | 'second'
```

Comment (tan75) [reply src/formatDistanceStrict/index.ts:170]: As well, could you please change the remaining `var`s.
Many thanks!

**Review by tan75 — CHANGES_REQUESTED** [review]
Thanks a lot for this PR!
Please address my comments!

File: src/formatDistanceStrict/index.ts (hunk 1 above)
Comment (a-korzun) [reply src/formatDistanceStrict/index.ts:170]: Yeah, sure! Will do it asap

**Review by tan75 — APPROVED** [review]
Thank you!

**Pull request merged**